sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/kafka/AzureEventHubsKafkaAutoConfiguration.java

```diff
@@ -64,29 +61,17 @@ public StaticConnectionStringProvider<AzureServiceType.EventHubs> eventHubsKafka
     }
 
     /**
-     * The Azure {@link KafkaProperties} instance will be created if an Azure Event Hubs connection string is provided
+     * The {@link KafkaPropertiesBeanPostProcessor} will be created if an Azure Event Hubs connection string is provided
      * and the Kafka dependency is detected from the classpath.
      * @param connectionStringProvider the Azure Event Hubs connection string provider.
-     * @return the {@link KafkaProperties} with an Azure Event Hubs connection information.
+     * @return the {@link KafkaPropertiesBeanPostProcessor} to configure Azure Event Hubs connection information
+     * to {@link KafkaProperties}.
      */
-    @Primary
     @Bean
     @ConditionalOnBean(value = AzureServiceType.EventHubs.class, parameterizedContainer = ServiceConnectionStringProvider.class)
```

Comment:
do you still need this `ConditionalOnBean`?

Reply:
This Autoconfiguration is our legacy kafka support, which uses event hubs connection string as credentials. Thus I think it requires the connection string provider bean?